sources/xfscli/src/bin/xfs_fsck.rs

```diff
@@ -0,0 +1,174 @@
+                let mut root_flags = "-o ".to_string();
+                for file in files.split(' ') {
+                    if file.starts_with("root=") {
+                        root = file.strip_prefix("root=").unwrap().to_string();
+                    }
+                    if file.starts_with("rootflags=") {
+                        root_flags = root_flags
+                            .clone()
+                            .add(file.strip_prefix("rootflags=").unwrap());
+                    }
+                }
+                let path = Path::new(&root);
+                let metadata = path.metadata().expect("Could not fetch metadata");
+                let file_type = metadata.file_type();
+
+                let test_result = Path::new(&root).exists() && file_type.is_block_device();
+                if !test_result {
+                    let blkid_result = Command::new(BLKID)
+                        .args(["-t", &root, "-o", target])
+                        .output()
+                        .expect("Couldn't run blkid");
+                    root = String::from_utf8(blkid_result.stdout).unwrap();
+                }
+
+                let basename_dev = Path::new(target).file_name().unwrap();
+
+                let basename_root = Path::new(&root).file_name().unwrap();
+
+                if basename_dev == basename_root {
+                    let mount_result = Command::new(MOUNT)
+                        .args([target, "/tmp/repair_mnt", root_flags.as_str()])
+                        .output()
+                        .expect("Couldn't run mount");
+                    if !mount_result.status.success() {
+                        exit(1);
+                    }
+                } else {
+                    let mount_result = Command::new(MOUNT)
+                        .args([target, "/tmp/repair_mnt"])
+                        .output()
+                        .expect("Couldn't run mount");
+                    if !mount_result.status.success() {
+                        exit(1);
+                    }
+                }
+
+                Command::new(UMOUNT)
+                    .args(["/tmp/repair_mnt"])
+                    .output()
+                    .expect("Couldn't run mount");
+
+                let xfs_repair = Command::new(XFS_REPAIR)
+                    .args(["-e", target])
+                    .output()
+                    .expect("Couldn't run xfs_repair again");
+                error = xfs_repair.status.code().unwrap();
+                fs::remove_dir("/tmp/repair_mnt").expect("could not delete");
+            }
+            exit(repair2fsck_code(error));
+        }
+
+        if auto {
+            println!("xfs fsck : XFS file system.");
+            exit(0);
+        } else {
+            println!(
+                "If you wish to check the consistency of an XFS filesystem or \
+            repair a damaged filesystem, see xfs_repair(8)."
+            );
+        }
```

Comment:
This script has very particular exit statuses that are relied upon by other tools to inform their own decisions. We should be very explicit about keeping those exit codes consistent.

Reply:
Resolved all the exit statuses.